Argo Ulisse 13 DCI support

kaschmo opened the pull request:
Provides support for Ulisse 13 DCI mobile split ACs
1. Created ir_Argo.h
2. Created ir_Argo.cpp
3. IRsend.h: added sendArgo(...)
4. IRremoteESP8266.h: 
    1. added #defines 
    2. added enum decode_type 
    3. added #define argo_message_length

kaschmo committed:
Create ir_Argo.h

Header file for Argo Ulisse 13 DCI AC support

kaschmo committed:
source for Argo Ulisse DCI 13 AC

kaschmo committed:
Update IRsend.h

kaschmo committed:
Update IRremoteESP8266.h

crankyoldgit committed:
Clean up a lot of lint and style issues

First pass at fixing a load of cpplint.py issues.

crankyoldgit committed:
Fix a number of style guide issues.

crankyoldgit committed:
Fix poor indenting.

crankyoldgit committed:
More style guide fixes.

crankyoldgit committed:
More cosmetic fixes

Style guide issues and spelling mistake.

crankyoldgit committed:
Style issue fixes

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+}
+
+void IRArgoESP::send() {
+  Serial.println("Sending IR code");
```
Please remove. We shouldn't be printing to serial from the library, except as a debug command.
e.g. DPRINTLN()

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+
+void IRArgoESP::send() {
+  Serial.println("Sending IR code");
+  _irsend.sendArgo(argo);
```
If you add a `checksum();` call before this, you can remove a whole bunch of other calls and save processing time.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+  // in MSB first
+  argo[9] = argo[9] | 0b00100000;  // Set ON/OFF bit to 1
+
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

kaschmo replied:
Took this pattern over from other A/C implementations. But since I am not storing the actual state in the bit array  you are right. Checksum is only required just before sending.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+  //in MSB first
+  //bit 5 of byte 9 to off
+  argo[9] = argo[9] & 0b11011111;  // Set on/off bit to 0
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+    argo[9] |= 0b00001000;
+  else
+    argo[9] &= 0b11110111;
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+
+  argo[2] += temp << 6;  // append to bit 6,7
+  argo[3] += temp >> 2;  // remove lowest to bits and append in 0-2
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+  argo[3] &= 0b11100111;
+  // Set fan mode at bit positions
+  argo[3] += fan << 3;
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+void IRArgoESP::setFlap(uint8_t flap) {
+  flap_mode = flap;
+  // TODO(kaschmo):
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+
+void IRArgoESP::setFlap(uint8_t flap) {
+  flap_mode = flap;
+  // TODO(kaschmo):
```
What is the TODO here?

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+
+  // Set cool mode at bit positions
+  argo[2] += mode << 3;
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+  argo[2] |= 0b00100000;
+  // Set cool mode at bit positions
+  argo[2] += mode << 3;
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+    argo[9] |= 0b00000100;
+  else
+    argo[9] &= 0b11111011;
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+    argo[9] |= 0b10000000;
+  else
+    argo[9] &= 0b01111111;
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+
+  argo[3] += temp << 5;  // Append to bit 5,6,7
+  argo[4] += temp >> 3;  // Remove lowest 3 bits and append in 0,1
+  checksum();
```
If you add this to where the other comments say, you can remove this instance of `checksum()`.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+/*
+Node MCU/ESP8266 Sketch to emulate Argo Ulisse 13 DCI remote
```
Can you please also add what models the remote/code is known to control. Just the one you have access to and tested against is fine for starters.

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,282 @@
+// Args:
+//   data: An array of ARGO_COMMAND_LENGTH bytes containing the IR command.
+//
+// Status: STABLE
```
BETA at best. ;-) We probably need others to confirm it's working before we can count it as stable. Or have some unit tests for it.

crankyoldgit reviewed (changes requested):
Excellent stuff! Thanks.

Can you also some example code, similar to the other A/Cs so people know how to use your addition?

FYI, this fails mostly due to a load of style/lint errors in Travis. (The changes I reviewed - https://travis-ci.org/markszabo/IRremoteESP8266/jobs/256445911)
We try to follow the Google C++ style guide. https://google.github.io/styleguide/cppguide.html
There is also a linter available for it too. https://github.com/google/styleguide/tree/gh-pages/cpplint

The linter is part of the Travis checks and will indicate an error in the build of the library.

Check  https://travis-ci.org/markszabo/IRremoteESP8266 for later updates etc.
I've tried to fix most of them for you.

e.g.
```
3.89s$ python cpplint.py --extensions=c,cc,cpp,ino --headers=h,hpp {src,test}/*.{h,c,cc,cpp,hpp,ino} examples/*/*.{h,c,cc,cpp,hpp,ino}
src/ir_Argo.h:124:  #endif line should be "#endif  // IR_ARGO_H_"  [build/header_guard] [5]
src/ir_Argo.h:32:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:32:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:33:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:33:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:34:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:34:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:35:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:35:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:37:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:37:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:38:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:38:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:39:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:39:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:42:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:42:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:43:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:43:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:45:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:45:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:46:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:46:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:47:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:47:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:48:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:48:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:50:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:50:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:51:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:51:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:52:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:52:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:53:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:53:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:54:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:54:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:55:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:55:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:56:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:56:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:57:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:57:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:105:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:105:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:108:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:108:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:110:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:115:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:115:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:118:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:118:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:119:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:119:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.h:120:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.h:120:  Should have a space between // and comment  [whitespace/comments] [4]
Done processing src/ir_Argo.h
Done processing src/ir_Daikin.h
Done processing src/ir_Fujitsu.h
Done processing src/ir_Kelvinator.h
Done processing src/ir_LG.h
Done processing src/ir_Mitsubishi.h
Done processing src/IRrecv.h
src/IRremoteESP8266.h:128:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/IRremoteESP8266.h:128:  Should have a space between // and comment  [whitespace/comments] [4]
src/IRremoteESP8266.h:129:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/IRremoteESP8266.h:129:  Should have a space between // and comment  [whitespace/comments] [4]
Done processing src/IRremoteESP8266.h
src/IRsend.h:178:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/IRsend.h:178:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
Done processing src/IRsend.h
Done processing src/IRtimer.h
Done processing src/IRutils.h
Done processing src/ir_Aiwa.cpp
src/ir_Argo.cpp:16:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:17:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:17:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:18:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:18:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:31:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:43:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:44:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:48:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:49:  Redundant blank line at the end of a code block should be deleted.  [whitespace/blank_line] [3]
src/ir_Argo.cpp:67:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:67:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:70:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:71:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:75:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:75:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:75:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:76:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:77:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:79:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:79:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:80:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:80:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:81:  Redundant blank line at the end of a code block should be deleted.  [whitespace/blank_line] [3]
src/ir_Argo.cpp:89:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:90:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:91:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:91:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:92:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:92:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:93:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:94:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:94:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:95:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:95:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:113:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:114:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:115:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:116:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:116:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:123:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:124:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:125:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:126:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:126:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:138:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:143:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:156:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:163:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:164:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:165:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:166:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:167:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:168:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:169:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:173:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:173:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:174:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:174:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:186:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:188:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:199:  Missing username in TODO; it should look like "// TODO(my_username): Stuff."  [readability/todo] [2]
src/ir_Argo.cpp:199:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:208:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:212:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:212:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:212:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:213:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:214:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:217:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:220:  Redundant blank line at the end of a code block should be deleted.  [whitespace/blank_line] [3]
src/ir_Argo.cpp:227:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:227:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:227:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:228:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:229:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:231:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:233:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:242:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:244:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:255:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:257:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:268:  Missing username in TODO; it should look like "// TODO(my_username): Stuff."  [readability/todo] [2]
src/ir_Argo.cpp:268:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:269:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:272:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:273:  Missing spaces around =  [whitespace/operators] [4]
src/ir_Argo.cpp:274:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:278:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:278:  Should have a space between // and comment  [whitespace/comments] [4]
src/ir_Argo.cpp:279:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/ir_Argo.cpp:279:  Should have a space between // and comment  [whitespace/comments] [4]
```

crankyoldgit committed:
Clean up indent/whitespace issue

crankyoldgit committed:
More style/lint fixes

crankyoldgit commented on src/ir_Argo.cpp:
```diff
@@ -0,0 +1,281 @@
+#include <Arduino.h>
+
+
+//  ARgo
```
Typo?

kaschmo committed:
Update ir_Argo.cpp

Cleaned up checksum calls in every set. Now called just before sending in send() and in getRaw().

crankyoldgit committed:
Fix lint issue.

crankyoldgit committed:
Fix lint issue.

kaschmo committed:
Merge branch 'master' into patch-1

kaschmo commented:
Excellent.
